e2e/bitrise.yml

```diff
@@ -0,0 +1,128 @@
+format_version: 11
+default_step_lib_source: https://github.com/bitrise-io/bitrise-steplib.git
+
+app:
+  envs:
+    - ORIG_BITRISE_SOURCE_DIR: $BITRISE_SOURCE_DIR
+
+workflows:
+  test_cache:
+    before_run:
+    - _setup_workdir
+    steps:
+    - git::https://github.com/bitrise-steplib/bitrise-step-simple-git-clone.git:
+        inputs:
+        - repository_url: https://github.com/bitrise-samples/sample-apps-react-native-ios-and-android
+        - branch: master
+        - clone_into_dir: ./_tmp
+    - change-workdir:
+        inputs:
+        - path: ./_tmp
+    - path::./:
+        title: Test with system provided npm
+        inputs:
+        - command: install
+        - cache_local_deps: true
+    - script:
+        title: Check if required files added to the cache env
+        inputs:
+        - content: |
+            #!/bin/bash
+            set -ex
+            if [[ $BITRISE_CACHE_INCLUDE_PATHS != *"$BITRISE_SOURCE_DIR/node_modules -> $BITRISE_SOURCE_DIR/package-lock.json"* ]]; then
+                echo "cache path not present"
+                exit 1
+            fi
+    - change-workdir:
+        inputs:
+        - path: $ORIG_BITRISE_SOURCE_DIR
+
+  test_system_npm_version:
+    before_run:
+    - _setup_workdir
+    steps:
+    - git::https://github.com/bitrise-steplib/bitrise-step-simple-git-clone.git:
+        inputs:
+        - repository_url: https://github.com/bitrise-samples/sample-apps-react-native-ios-and-android
+        - branch: master
+        - clone_into_dir: ./_tmp
+    - path::./:
+        title: Test with system provided npm
+        inputs:
+        - workdir: ./_tmp
+        - command: install
+        - cache_local_deps: true
+
+  test_packageJSON_npm_version:
+    before_run:
+    - _setup_workdir
+    steps:
+    - git::https://github.com/bitrise-steplib/bitrise-step-simple-git-clone.git:
+        inputs:
+        - repository_url: https://github.com/bitrise-samples/sample-apps-react-native-ios-and-android
+        - branch: master
+        - clone_into_dir: ./_tmp
+    - change-workdir:
+        inputs:
+        - path: ./_tmp
+    - script:
+        title: Generate npm version specifier into package.json
+        inputs:
+        - content: |
+            #!/bin/bash
+            # insert npm version specifier block into package.json
+            mv package.json package.json.old
+            awk '/SampleAppsReactNativeAndroid/{print "\"engines\":{\"npm\":\"6.14.6\"},"}1' package.json.old > package.json
+            echo "generated package.json:"
+            cat package.json
+    - path::./:
```

Comment:
Here too we should introduce the `_run` wf and the `test_` wfs just configure env-vars

Reply:
In this step it doesn't make sense as almost all tests are different.